src/Illuminate/Support/Collection.php

```diff
@@ -870,6 +870,19 @@ protected function valueRetriever($value)
     }
 
     /**
+     * Walk over each of the items.
+     *
+     * @param  callable  $callback
+     * @return static
+     */
+    public function walk(callable $callback)
+    {
+        array_walk($this->items, $callback);
```

Comment:
I think it doesn't look right because, in the case of `array_walk`, there isn't a reason to create a new Collection instance. Part of the use case for why I needed something like this is I needed to be able to modify items in a collection directly. I added the `return new static($this->items);` bit to try to keep with the common behavior that other similar methods provide like `map()` and `filter()`. Do you think I should instead just return `$this`?